### PR TITLE
add autoflake and move configs to pyproject.toml

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Testing config updates before adding to the project template

Related to:
https://github.com/ethereum/ethereum-python-project-template/issues/95
and
https://github.com/ethereum/ethereum-python-project-template/issues/39

### How was it fixed?
 - Added `autoflake` to linting - removing all unused imports in any file not explicitly excluded, and remove unnecessary `pass` statements.
 - Move all possible config over to `pyproject.toml`. This is all of our linting tools except for `flake8`.
 - Removed `pytest-watch` from dev dependencies, not looking currently maintained.

### Todo:

- [x] Clean up commit history

- [x] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/pyrlp/blob/main/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

**Review by reedsa — APPROVED** [review]
Seems legit, :shipit:

**Review by fselmo — APPROVED** [review]
lgtm... if we're copying this over to the template I just added a spacing comment to get it all similar.

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,71 @@
...
+
+# Explanation:
+# D400 - Enabling this error code seems to make it a requirement that the first
+# sentence in a docstring is not split across two lines.  It also makes it a
+# requirement that no docstring can have a multi-sentence description without a
+# summary line.  Neither one of those requirements seem appropriate.
+
+[tool.pytest.ini_options]
+addopts= "-v --showlocals --durations 10"
+xfail_strict= "True"
```
[the remark this answers is not in the document]
Comment (fselmo) [reply pyproject.toml:64]: oh `addopts` too

[the remark this answers is not in the document]
Comment (pacrob) [reply pyproject.toml]: fixed!

**Pull request merged**